authentication/securestore/main.go

```diff
@@ -103,32 +118,42 @@ func UpsertConfigToKeyring(creds []byte) error {
 // RemoveConfigFromKeyring will remove the credentials from the first priority OS secure store.
 //
 // Note: CGO_ENABLED=1 is required for OSX Keychain and darwin builds
-func RemoveConfigFromKeyring() error {
-	if err := validateBackends(); err != nil {
+func RemoveConfigFromKeyring(backend string) error {
+	if err := ValidateBackend(backend); err != nil {
 		return err
 	}
 
-	ring, err := keyring.Open(getKeyringConfig())
+	ring, err := keyring.Open(getKeyringConfig(backend))
 	if err != nil {
 		return err
 	}
 
 	err = ring.Remove(ItemKey)
+	if err != nil {
+		if err == keyring.ErrKeyNotFound {
+			// Ignore not found errors, key is already removed
+			return nil
+		}
+
+		if strings.Contains(err.Error(), "Keychain Error. (-25244)") {
+			return fmt.Errorf("%s\nThis application may not have permission to delete from the Keychain. Please check the permissions in the Keychain and try again", err.Error())
```

Comment:
Friendly reminder that when wrapping errors your should use "%w" rather than %s or %v to properly preserve the original error in case callers want it: https://gosamples.dev/wrap-unwrap-errors/